ai-service/src/ai_service/errors.py

```
@@ -38,3 +38,31 @@ def empty_embedding(cls) -> "InvalidParam":
+        return cls(f"File not found: {file_path}")
+
+    @classmethod
+    def permission_denied(cls, file_path: str) -> "FileReadError":
+        return cls(f"Permission denied: {file_path}")
+
+    @classmethod
+    def decode_error(cls, file_path: str) -> "FileReadError":
+        return cls(f"Unicode decode error in file: {file_path}")
+
+    @classmethod
+    def os_error(cls, file_path: str, error) -> "FileReadError":
```

Ruff (0.12.2)

Missing type annotation for function argument `error`

(ANN001)